# Udemy 강의

## Section 4. 판다스 시리즈와 인덱스 객체

### **판다스 시리즈 만들기**

In [1]:
import pandas as pd

In [2]:
summer = pd.read_csv(r'C:\Users\A\Desktop\데잇걸즈\3. python\udemy\Course_Materials_Part1\Video_Lecture_NBs\summer.csv')

In [3]:
summer.head()

Year    City     Sport Discipline             Athlete Country Gender  \
0  1896  Athens  Aquatics   Swimming       HAJOS, Alfred     HUN    Men   
1  1896  Athens  Aquatics   Swimming    HERSCHMANN, Otto     AUT    Men   
2  1896  Athens  Aquatics   Swimming   DRIVAS, Dimitrios     GRE    Men   
3  1896  Athens  Aquatics   Swimming  MALOKINIS, Ioannis     GRE    Men   
4  1896  Athens  Aquatics   Swimming  CHASAPIS, Spiridon     GRE    Men   

                        Event   Medal  
0              100M Freestyle    Gold  
1              100M Freestyle  Silver  
2  100M Freestyle For Sailors  Bronze  
3  100M Freestyle For Sailors    Gold  
4  100M Freestyle For Sailors  Silver

- **시리즈 만들기 1**

In [7]:
summer["Athlete"]

0                    HAJOS, Alfred
1                 HERSCHMANN, Otto
2                DRIVAS, Dimitrios
3               MALOKINIS, Ioannis
4               CHASAPIS, Spiridon
                   ...            
31160           JANIKOWSKI, Damian
31161    REZAEI, Ghasem Gholamreza
31162               TOTROV, Rustam
31163            ALEKSANYAN, Artur
31164               LIDBERG, Jimmy
Name: Athlete, Length: 31165, dtype: object

- **시리즈 만들기 2**

In [5]:
summer.Athlete

0                    HAJOS, Alfred
1                 HERSCHMANN, Otto
2                DRIVAS, Dimitrios
3               MALOKINIS, Ioannis
4               CHASAPIS, Spiridon
                   ...            
31160           JANIKOWSKI, Damian
31161    REZAEI, Ghasem Gholamreza
31162               TOTROV, Rustam
31163            ALEKSANYAN, Artur
31164               LIDBERG, Jimmy
Name: Athlete, Length: 31165, dtype: object

- **시리즈 만들기 3**

In [6]:
summer.iloc[0]

Year                    1896
City                  Athens
Sport               Aquatics
Discipline          Swimming
Athlete        HAJOS, Alfred
Country                  HUN
Gender                   Men
Event         100M Freestyle
Medal                   Gold
Name: 0, dtype: object

- **원하는 열만 선택해서 데이터 임포트하기**
- **usecols에 임포트하고싶은 열을 입력. 여러 열의 이름을 리스트로 넣을 수 있음**
- **squeeze = True면 usecols에 하나의 열만 입력됐을 때 시리즈로 반환한다는 의미**

In [8]:
pd.read_csv(r'C:\Users\A\Desktop\데잇걸즈\3. python\udemy\Course_Materials_Part1\Video_Lecture_NBs\summer.csv', usecols = ["Athlete"], squeeze = True).head()

0         HAJOS, Alfred
1      HERSCHMANN, Otto
2     DRIVAS, Dimitrios
3    MALOKINIS, Ioannis
4    CHASAPIS, Spiridon
Name: Athlete, dtype: object

- **처음부터 시리즈를 만들고 싶을 때 pd.Series() 사용**

- **데이터 값과 인덱스의 개수가 동일해야 함에 주의**

In [10]:
pd.Series([10,25,6,36,2], index=["Mon","Tue","Wed","Thu","Fri"])

Mon    10
Tue    25
Wed     6
Thu    36
Fri     2
dtype: int64

- **시리즈의 이름 붙여주기**

In [11]:
pd.Series([10,25,6,36,2], index=["Mon","Tue","Wed","Thu","Fri"], name = "Sales")

Mon    10
Tue    25
Wed     6
Thu    36
Fri     2
Name: Sales, dtype: int64

- **배열 만들고 시리즈로 변환하기**

In [12]:
import numpy as np

In [13]:
sales = np.array([10,25,6,36,2])
sales

array([10, 25,  6, 36,  2])

In [14]:
pd.Series(sales)

0    10
1    25
2     6
3    36
4     2
dtype: int32

- **pd.Series() 안에 리스트를 바로 넣을 수 있음**
- **sales 변수를 리스트로 만들어서 pd.Series(sales)로 넣어도 시리즈 출력**
- **튜플로 해주어도 결과는 동일**

In [15]:
sales = list(sales)

In [16]:
pd.Series(sales)

0    10
1    25
2     6
3    36
4     2
dtype: int64

- **딕셔너리를 시리즈로 변환하기**

In [17]:
dic = {"Mon":10, "Tue":25, "Wed":6, "Thu":36, "Fri":2}
dic

{'Mon': 10, 'Tue': 25, 'Wed': 6, 'Thu': 36, 'Fri': 2}

In [18]:
sales = pd.Series(dic)
sales

Mon    10
Tue    25
Wed     6
Thu    36
Fri     2
dtype: int64

- **dic에서 key값을 통해 인덱스를 지정해주었기 때문에 pd.Series에 인덱스를 정해주면 충돌됨**
- **인덱스 순서를 변경하고 Sat, Sun 추가, Thu는 제외되었을 때 dic값과 비교해 기존 key와 맞는 값을 가져오고 Sat, Sun을 인덱스로 가지는 값은 NaN으로 추가됨**
- **Thu의 값은 사라진 것을 확인**

In [19]:
pd.Series(dic, index = ["Fri", "Sat", "Sun", "Mon", "Tue", "Wed"])

Fri     2.0
Sat     NaN
Sun     NaN
Mon    10.0
Tue    25.0
Wed     6.0
dtype: float64

- **기존 인덱스와 다른 새로운 인덱스 넣어보기**
- **원래 딕셔너리에 해당하는 key값이 없기 때문에 모두 NaN 반환**

In [20]:
pd.Series(dic, index = [1,2,3,4,5])

1   NaN
2   NaN
3   NaN
4   NaN
5   NaN
dtype: float64

### **시리즈의 인덱싱과 슬라이싱**

In [21]:
titanic = pd.read_csv(r'C:\Users\A\Desktop\데잇걸즈\3. python\udemy\Course_Materials_Part1\Video_Lecture_NBs\titanic.csv')

In [22]:
titanic.head()

survived  pclass     sex   age  sibsp  parch     fare embarked deck
0         0       3    male  22.0      1      0   7.2500        S  NaN
1         1       1  female  38.0      1      0  71.2833        C    C
2         1       3  female  26.0      0      0   7.9250        S  NaN
3         1       1  female  35.0      1      0  53.1000        S    C
4         0       3    male  35.0      0      0   8.0500        S  NaN

In [23]:
age = titanic.age

In [24]:
age.head()

0    22.0
1    38.0
2    26.0
3    35.0
4    35.0
Name: age, dtype: float64

In [28]:
age.tail()

886    27.0
887    19.0
888     NaN
889    26.0
890    32.0
Name: age, dtype: float64

In [25]:
age.index

RangeIndex(start=0, stop=891, step=1)

- **대괄호를 이용하면 파이썬이 내부적으로 순서에 기반한 인덱싱을 하는지 인덱스 라벨에 기반한 인덱싱을 하는지 알기 어려움**

In [26]:
age[0]

22.0

- **순서 기반 인덱싱이라면 -1을 넣었을 때 마지막 행 값이 나옴**
- **오류를 확인해보면 해당 인덱싱은 순서 기반 인덱싱이 아닌 인덱스 라벨에 기반한 것임을 알 수 있음**

In [29]:
age[-1]

KeyError: -1

- **iloc를 이용해 순서 기반 인덱싱이 이뤄지는 것을 볼 수 있음**

In [30]:
age.iloc[-1]

32.0

In [31]:
age[[3,4]]

3    35.0
4    35.0
Name: age, dtype: float64

In [32]:
age[:3]

0    22.0
1    38.0
2    26.0
Name: age, dtype: float64

- **대괄호를 이용한 슬라이싱이 age[-1]과 같은 순서에 기반하지 않은 인덱싱과 다르게 순서에 기반한 것과 같은 결과가 나옴**
- **결국, RangeIndex로 인덱스가 있을 경우, 대괄호를 이용한 슬라이싱이나 인덱싱은 순서에 기반한 것인지 인덱스 라벨에 기반한 것인지 알기 어려움**
- **따라서 혼란을 피하기 위해 iloc 연산자를 이용한 인덱싱이 더 좋다고 볼 수 있음**

In [33]:
age.iloc[:3]

0    22.0
1    38.0
2    26.0
Name: age, dtype: float64

- **iloc 대신 loc 연산자를 사용하면 라벨 기반 인덱싱을 할 수 있음**
- **loc를 사용할 때는 시작 인덱스와 마지막 인덱스를 포함**

In [34]:
age.loc[:3]

0    22.0
1    38.0
2    26.0
3    35.0
Name: age, dtype: float64

- **순서 기반 인덱싱 : iloc**
- **라벨 기반 인덱싱 : loc**

In [35]:
summer = pd.read_csv(r'C:\Users\A\Desktop\데잇걸즈\3. python\udemy\Course_Materials_Part1\Video_Lecture_NBs\summer.csv', index_col = "Athlete")

In [36]:
summer.head()

Year    City     Sport Discipline Country Gender  \
Athlete                                                                
HAJOS, Alfred       1896  Athens  Aquatics   Swimming     HUN    Men   
HERSCHMANN, Otto    1896  Athens  Aquatics   Swimming     AUT    Men   
DRIVAS, Dimitrios   1896  Athens  Aquatics   Swimming     GRE    Men   
MALOKINIS, Ioannis  1896  Athens  Aquatics   Swimming     GRE    Men   
CHASAPIS, Spiridon  1896  Athens  Aquatics   Swimming     GRE    Men   

                                         Event   Medal  
Athlete                                                 
HAJOS, Alfred                   100M Freestyle    Gold  
HERSCHMANN, Otto                100M Freestyle  Silver  
DRIVAS, Dimitrios   100M Freestyle For Sailors  Bronze  
MALOKINIS, Ioannis  100M Freestyle For Sailors    Gold  
CHASAPIS, Spiridon  100M Freestyle For Sailors  Silver

In [37]:
event = summer.Event

In [38]:
event.head()

Athlete
HAJOS, Alfred                     100M Freestyle
HERSCHMANN, Otto                  100M Freestyle
DRIVAS, Dimitrios     100M Freestyle For Sailors
MALOKINIS, Ioannis    100M Freestyle For Sailors
CHASAPIS, Spiridon    100M Freestyle For Sailors
Name: Event, dtype: object

In [39]:
event.tail()

Athlete
JANIKOWSKI, Damian           Wg 84 KG
REZAEI, Ghasem Gholamreza    Wg 96 KG
TOTROV, Rustam               Wg 96 KG
ALEKSANYAN, Artur            Wg 96 KG
LIDBERG, Jimmy               Wg 96 KG
Name: Event, dtype: object

- **임포트할 때 인덱스를 지정해줬기 때문에 RangeIndex가 아닌 선수 이름으로 된 것을 볼 수 있음**

In [40]:
event.index

Index(['HAJOS, Alfred', 'HERSCHMANN, Otto', 'DRIVAS, Dimitrios',
       'MALOKINIS, Ioannis', 'CHASAPIS, Spiridon', 'CHOROPHAS, Efstathios',
       'HAJOS, Alfred', 'ANDREOU, Joannis', 'CHOROPHAS, Efstathios',
       'NEUMANN, Paul',
       ...
       'AHMADOV, Emin', 'KAZAKEVIC, Aleksandr', 'KHUGAEV, Alan',
       'EBRAHIM, Karam Mohamed Gaber', 'GAJIYEV, Danyal', 'JANIKOWSKI, Damian',
       'REZAEI, Ghasem Gholamreza', 'TOTROV, Rustam', 'ALEKSANYAN, Artur',
       'LIDBERG, Jimmy'],
      dtype='object', name='Athlete', length=31165)

In [41]:
event[0]

'100M Freestyle'

- **인덱스가 RangeIndex가 아니기 때문에 -1이 값의 순서인 것을 알게 되어 순서 기반 인덱싱인 -1도 작동함**

In [42]:
event[-1]

'Wg 96 KG'

In [44]:
event.iloc[:3]

Athlete
HAJOS, Alfred                    100M Freestyle
HERSCHMANN, Otto                 100M Freestyle
DRIVAS, Dimitrios    100M Freestyle For Sailors
Name: Event, dtype: object

In [45]:
event["DRIVAS, Dimitrios"]

'100M Freestyle For Sailors'

In [46]:
event[:"DRIVAS, Dimitrios"]

Athlete
HAJOS, Alfred                    100M Freestyle
HERSCHMANN, Otto                 100M Freestyle
DRIVAS, Dimitrios    100M Freestyle For Sailors
Name: Event, dtype: object

In [48]:
event.loc["PHELPS, Michael"]

Athlete
PHELPS, Michael            100M Butterfly
PHELPS, Michael            200M Butterfly
PHELPS, Michael            200M Freestyle
PHELPS, Michael    200M Individual Medley
PHELPS, Michael    400M Individual Medley
PHELPS, Michael    4X100M Freestyle Relay
PHELPS, Michael       4X100M Medley Relay
PHELPS, Michael    4X200M Freestyle Relay
PHELPS, Michael            100M Butterfly
PHELPS, Michael            200M Butterfly
PHELPS, Michael            200M Freestyle
PHELPS, Michael    200M Individual Medley
PHELPS, Michael    400M Individual Medley
PHELPS, Michael    4X100M Freestyle Relay
PHELPS, Michael       4X100M Medley Relay
PHELPS, Michael    4X200M Freestyle Relay
PHELPS, Michael            100M Butterfly
PHELPS, Michael            200M Butterfly
PHELPS, Michael               200M Medley
PHELPS, Michael          4X100M Freestyle
PHELPS, Michael             4X100M Medley
PHELPS, Michael          4X200M Freestyle
Name: Event, dtype: object

- **중복되는 인덱스를 가지고 슬라이싱을 하면 오류 발생**
- **파이썬이 어떤 PHELPS, Michael을 의미하는지 알지 못하기 때문**

In [49]:
event.loc[:"PHELPS, Michael"]

KeyError: "Cannot get right slice bound for non-unique label: 'PHELPS, Michael'"

In [52]:
event.loc[["PHELPS, Michael", "LEWIS, Carl"]]

Athlete
PHELPS, Michael            100M Butterfly
PHELPS, Michael            200M Butterfly
PHELPS, Michael            200M Freestyle
PHELPS, Michael    200M Individual Medley
PHELPS, Michael    400M Individual Medley
PHELPS, Michael    4X100M Freestyle Relay
PHELPS, Michael       4X100M Medley Relay
PHELPS, Michael    4X200M Freestyle Relay
PHELPS, Michael            100M Butterfly
PHELPS, Michael            200M Butterfly
PHELPS, Michael            200M Freestyle
PHELPS, Michael    200M Individual Medley
PHELPS, Michael    400M Individual Medley
PHELPS, Michael    4X100M Freestyle Relay
PHELPS, Michael       4X100M Medley Relay
PHELPS, Michael    4X200M Freestyle Relay
PHELPS, Michael            100M Butterfly
PHELPS, Michael            200M Butterfly
PHELPS, Michael               200M Medley
PHELPS, Michael          4X100M Freestyle
PHELPS, Michael             4X100M Medley
PHELPS, Michael          4X200M Freestyle
LEWIS, Carl                          100M
LEWIS, Carl               

### **시리즈 정렬하기 & inplace 파라미터**

In [53]:
dic = {1:10, 3:25, 2:6, 4:36, 5:2, 6:0, 7:None}
dic

{1: 10, 3: 25, 2: 6, 4: 36, 5: 2, 6: 0, 7: None}

In [54]:
sales = pd.Series(dic)
sales

1    10.0
3    25.0
2     6.0
4    36.0
5     2.0
6     0.0
7     NaN
dtype: float64

- **제대로 정렬되지 않은 위의 결과와 달리 sort_index() 후에 제대로 정렬된 것을 확인**
- **sort_index()의 디폴트는 ascending = True**

In [55]:
sales.sort_index()

1    10.0
2     6.0
3    25.0
4    36.0
5     2.0
6     0.0
7     NaN
dtype: float64

In [56]:
sales.sort_index(ascending=False)

7     NaN
6     0.0
5     2.0
4    36.0
3    25.0
2     6.0
1    10.0
dtype: float64

In [57]:
sales.sort_index(ascending=True)

1    10.0
2     6.0
3    25.0
4    36.0
5     2.0
6     0.0
7     NaN
dtype: float64

- **sales를 확인해보면 아직 정렬이 반영되지 않음. sales의 복사본을 만들어 복사본에 정렬한 것**

In [58]:
sales

1    10.0
3    25.0
2     6.0
4    36.0
5     2.0
6     0.0
7     NaN
dtype: float64

- **inplace = True면 정렬된 값이 원래 값을 대신하게 됨**

In [59]:
sales.sort_index(ascending = True, inplace = True)

In [60]:
sales

1    10.0
2     6.0
3    25.0
4    36.0
5     2.0
6     0.0
7     NaN
dtype: float64

- **sort_values를 이용해 값을 기준으로 정렬하기**

In [61]:
sales.sort_values()

6     0.0
5     2.0
2     6.0
1    10.0
3    25.0
4    36.0
7     NaN
dtype: float64

- **na_position을 이용해 NaN 값의 위치를 정해줄 수도 있음**

In [63]:
sales.sort_values(ascending=False, na_position="last", inplace=True)

In [64]:
sales

4    36.0
3    25.0
1    10.0
2     6.0
5     2.0
6     0.0
7     NaN
dtype: float64

- **sort_index로 정렬할 때, 인덱스가 문자열이라면 알파벳 순으로 정렬**

In [65]:
dic = {"Mon":10, "Tue":25, "Wed":6, "Thu":36, "Fri":2}
dic

{'Mon': 10, 'Tue': 25, 'Wed': 6, 'Thu': 36, 'Fri': 2}

In [66]:
sales = pd.Series(dic)

In [67]:
sales

Mon    10
Tue    25
Wed     6
Thu    36
Fri     2
dtype: int64

In [68]:
sales.sort_index(ascending=True)

Fri     2
Mon    10
Thu    36
Tue    25
Wed     6
dtype: int64

### **nlargest() & nsmallest()**

In [69]:
titanic = pd.read_csv(r'C:\Users\A\Desktop\데잇걸즈\3. python\udemy\Course_Materials_Part1\Video_Lecture_NBs\titanic.csv')

In [70]:
titanic.head()

survived  pclass     sex   age  sibsp  parch     fare embarked deck
0         0       3    male  22.0      1      0   7.2500        S  NaN
1         1       1  female  38.0      1      0  71.2833        C    C
2         1       3  female  26.0      0      0   7.9250        S  NaN
3         1       1  female  35.0      1      0  53.1000        S    C
4         0       3    male  35.0      0      0   8.0500        S  NaN

In [71]:
age = titanic.age

In [73]:
age.head()

0    22.0
1    38.0
2    26.0
3    35.0
4    35.0
Name: age, dtype: float64

- **age에서 나이가 가장 많은 세 승객 알아보기**

In [75]:
age.sort_values(ascending=False).head()

630    80.0
851    74.0
493    71.0
96     71.0
116    70.5
Name: age, dtype: float64

- **nlargest()를 이용하면 시리즈의 가장 큰 값 5개를 불러옴**

In [76]:
age.nlargest()

630    80.0
851    74.0
96     71.0
493    71.0
116    70.5
Name: age, dtype: float64

- **가장 어린 승객 3명 알아보기**

In [78]:
age.sort_values(ascending=True).head()

803    0.42
755    0.67
644    0.75
469    0.75
78     0.83
Name: age, dtype: float64

- **nsmallest를 이용하면 시리즈의 가장 작은 값 5개를 불러옴**

In [80]:
age.nsmallest()

803    0.42
755    0.67
469    0.75
644    0.75
78     0.83
Name: age, dtype: float64

### **idxmin() & idxmax()**

- **최댓값, 최솟값의 인덱스 라벨만 보기**

In [81]:
age.nlargest(n=3).index[0]

630

In [82]:
age.nsmallest(n=3).index[0]

803

- **idxmax(), idxmin()을 이용해 구하기**

In [83]:
titanic.age.idxmax()

630

In [84]:
titanic.age.idxmin()

803

- **제일 나이가 어린/많은 승객의 전체 정보는 loc 연산자를 이용해 확인할 수 있음**

In [85]:
titanic.loc[630]

survived       1
pclass         1
sex         male
age         80.0
sibsp          0
parch          0
fare        30.0
embarked       S
deck           A
Name: 630, dtype: object

- **가장 나이가 어린 승객의 인덱스를 구해서 전체 정보를 불러오는 과정을 하나로 합치기**

In [86]:
titanic.loc[titanic.age.idxmin()]

survived         1
pclass           3
sex           male
age           0.42
sibsp            0
parch            1
fare        8.5167
embarked         C
deck           NaN
Name: 803, dtype: object

In [87]:
dic = dic = {"Mon":10, "Tue":25, "Wed":6, "Thu":36, "Fri":2, "Sat":0, "Sun":None}
dic

{'Mon': 10, 'Tue': 25, 'Wed': 6, 'Thu': 36, 'Fri': 2, 'Sat': 0, 'Sun': None}

In [88]:
sales = pd.Series(dic)
sales

Mon    10.0
Tue    25.0
Wed     6.0
Thu    36.0
Fri     2.0
Sat     0.0
Sun     NaN
dtype: float64

- **판매량이 제일 적은 요일 찾아보기**

In [89]:
sales.sort_values(ascending=True).index[0]

'Sat'

In [90]:
sales.idxmin()

'Sat'

- **판매량이 제일 많았던 요일**

In [91]:
sales.sort_values(ascending=False).index[0]

'Thu'

In [92]:
sales.idxmax()

'Thu'

### **판다스 시리즈 다루기**

- **판다스 시리즈 안의 하나의 값을 바꾸기, 조작하기, 전체 시리즈를 변경하기, 덮어쓰기**

- **먼저 일간 매출 자료로 시리즈를 만들어주기**

In [94]:
sales = pd.Series([10,25,6,36,2,0,None,5], index = ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun", "Mon"])
sales

Mon    10.0
Tue    25.0
Wed     6.0
Thu    36.0
Fri     2.0
Sat     0.0
Sun     NaN
Mon     5.0
dtype: float64

- **null값인 Sun에 0값 집어넣기**

In [95]:
sales["Sun"] = 0

In [96]:
sales

Mon    10.0
Tue    25.0
Wed     6.0
Thu    36.0
Fri     2.0
Sat     0.0
Sun     0.0
Mon     5.0
dtype: float64

- **인덱싱을 이용해 값 변경하기**

In [97]:
sales.iloc[3] = 30
sales

Mon    10.0
Tue    25.0
Wed     6.0
Thu    30.0
Fri     2.0
Sat     0.0
Sun     0.0
Mon     5.0
dtype: float64

- **판다스 시리즈에서는 Numpy와 같이 데이터끼리 벡터화연산 가능**

- **판매량 데이터가 달러 기준일 때 유로화로 바꾸기**

In [98]:
(sales/1.1)

Mon     9.090909
Tue    22.727273
Wed     5.454545
Thu    27.272727
Fri     1.818182
Sat     0.000000
Sun     0.000000
Mon     4.545455
dtype: float64

In [99]:
sales_EUR = (sales/1.1).round(2)
sales_EUR

Mon     9.09
Tue    22.73
Wed     5.45
Thu    27.27
Fri     1.82
Sat     0.00
Sun     0.00
Mon     4.55
dtype: float64

- **Mon의 값을 변경했을 때, 이전에는 9.09, 4.55유로였다면 지금은 모두 0으로 변경된 것 확인**
- **라벨 인덱싱을 사용해서 덮어쓰기를 하면 중복된 인덱스 라벨 값이 모두 변경**

In [100]:
sales["Mon"] = 0
sales

Mon     0.0
Tue    25.0
Wed     6.0
Thu    30.0
Fri     2.0
Sat     0.0
Sun     0.0
Mon     0.0
dtype: float64

In [101]:
titanic = pd.read_csv(r'C:\Users\A\Desktop\데잇걸즈\3. python\udemy\Course_Materials_Part1\Video_Lecture_NBs\titanic.csv')

In [102]:
age = titanic["age"]
age.head()

0    22.0
1    38.0
2    26.0
3    35.0
4    35.0
Name: age, dtype: float64

In [103]:
age.iloc[1] = 30

C:\Users\A\anaconda3\lib\site-packages\pandas\core\indexing.py:1637: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_single_block(indexer, value, name)


In [104]:
age.head()

0    22.0
1    30.0
2    26.0
3    35.0
4    35.0
Name: age, dtype: float64

In [105]:
titanic.head()

survived  pclass     sex   age  sibsp  parch     fare embarked deck
0         0       3    male  22.0      1      0   7.2500        S  NaN
1         1       1  female  30.0      1      0  71.2833        C    C
2         1       3  female  26.0      0      0   7.9250        S  NaN
3         1       1  female  35.0      1      0  53.1000        S    C
4         0       3    male  35.0      0      0   8.0500        S  NaN

- **데이터프레임에서 뽑아낸 시리즈에 값을 변경하면 기존 데이터프레임의 값도 같이 변경됨**
- **위의 경고메시지는 시리즈의 값을 변경하면 기존 데이터프레임 값도 변경된다는 것을 경고한 것**